File: src/main/kotlin/javaway/Task.java
package javaway;

import org.jetbrains.annotations.NotNull;

import java.time.Instant;

public record Task(
    int taskId,
    @NotNull String title,
    boolean isCompleted,
    @NotNull TaskType taskType,
    @NotNull Instant createdAt,
    @NotNull Instant modifiedAt
) {
    public Task {
        if (title.isEmpty()) {
            throw new IllegalArgumentException("Title of task can not be empty.");
        }
    }
};

File: src/main/kotlin/kotlinway/Task.kt
package ru.sinvic.kotlinway

import java.time.Instant

data class Task(
    val id: Int,
    val title: String,
    val isCompleted: Boolean = false,
    val type: TaskType = TaskType.PERSONAL,
    val createdAt: Instant = Instant.now(),
    val modifiedAt: Instant = Instant.now()
) {
    init {
        require(title.isNotBlank()) { "Title of task can not be empty." }
    }
}